packages/react-server/src/ReactFizzServer.js

```diff
@@ -520,6 +521,94 @@ function renderSuspenseBoundary(
   popComponentStackInDEV(task);
 }
 
+function renderSuspenseBoundaryWithoutFallback(
+  request: Request,
+  task: Task,
+  props: Object,
+) {
+  pushBuiltInComponentStackInDEV(task, 'Suspense');
+  const parentBoundary = task.blockedBoundary;
+  const parentSegment = task.blockedSegment;
+
+  // We need to push an "empty" thing here to identify the parent suspense boundary.
+  pushEmpty(parentSegment.chunks, request.responseState, task.assignID);
+  task.assignID = null;
+
+  const fallbackAbortSet: Set<Task> = new Set();
+  const content: ReactNodeList = props.children;
+
+  const newBoundary = createSuspenseBoundary(request, fallbackAbortSet);
```

Comment:
You don't actually need to create a Suspense boundary here. This only exists to handle things being "caught" and to write the comment nodes (using writeStartCompletedSuspenseBoundary).

If they're not caught, they just populate to the one above it. Any other type of component can throw at any time and it's just caught above it. So you can just not do any of that.

However, if you do that, then you wouldn't call writeStartCompletedSuspenseBoundary and miss out on writing the comments. However, you know that if it every completes, it'll be a completed boundary. So you can just add what writeStartCompletedSuspenseBoundary would do to the buffer directly.

You can't use the `write*` functions to add to a buffer. They write directly to the underlying stream.

I've used the `push*` convention for adding to a buffer.

You can fork writeStartCompletedSuspenseBoundary and writeEndCompletedSuspenseBoundary into a form that adds to a buffer. E.g. pushStartCompletedSuspenseBoundary and pushEndCompletedSuspenseBoundary. Look at what pushStartInstance and pushEndInstance does.

Then you can just write the comment nodes directly to the buffer here.